### ISSUE TITLE
Don't fail ARIA attributes reflected from an invalid to a valid element

### ISSUE THREAD

**Issue by WilcoFiers** [dequelabs/axe-core#5217]
Found this issue while benchmarking. The following shouldn't fail `aria-allowed-attr`:

```html
<text-field aria-required="true">
  <input aria-required="true" />
</text-field>
```

I think what to do here is, before Axe fails the element for having aria-required, check if any of its descendants has the same property AND value. 

There are a few things to consider when fixing this:

1. Which attributes do we want to allow this on? All non-global, or maybe a more limited set?
2. Should we only allow this on generics like?
3. Do we want this to go to needs review?
4. What happens if multiple descendants have this prop?
5. Should Axe limit how far it searches?
